ui/src/dashboards/actions/index.js

```diff
@@ -195,10 +195,26 @@ export const getDashboardsAsync = () => async dispatch => {
   }
 }
 
+const dashboardWithOnlySelectedTemplateValues = dashboard => {
+  const templates = dashboard.templates.map(template => {
+    const values =
+      template.type === 'csv'
+        ? template.values
+        : [template.values.find(val => val.selected)] || []
+    return {...template, values}
+  })
+  return templates
+}
+
 export const putDashboard = dashboard => async dispatch => {
   try {
-    const {data} = await updateDashboardAJAX(dashboard)
-    dispatch(updateDashboard(data))
+    // for server, template var values should be all values for csv
+    // and should be only the selected value for non csv types
+    const templates = dashboardWithOnlySelectedTemplateValues(dashboard)
+    const {data} = await updateDashboardAJAX({...dashboard, templates})
+    // updateDashboardAJAX removed the values for the template variables
+    // when saving to the server
+    dispatch(updateDashboard({...data, templates: dashboard.templates}))
```

Comment:
I find these comments a bit too verbose- Could you find variable names that explain what's going on? 
`dashboardWithOnlySelectedTemplateValues` could be `removeUnselectedTemplateValues(dashboard)` and `const {data: dashboardWithoutTemplates}` could be `{...dashboardWithoutTemplates, templates: dashboard.templates}))`